src/main/haskell/kore/test/Test/Kore/IndexedModule/MetadataTools.hs

```diff
@@ -146,26 +151,106 @@ test_metadataTools =
         (assertEqual ""
             []
             (applicationSortsOperands
-                (sortTools metadataTools (symbolHead objectA)))
+                (symbolOrAliasSorts metadataTools (symbolHead objectA)))
         )
     , testCase "getArgumentSorts meta"
         (assertEqual ""
             []
             (applicationSortsOperands
-                (sortTools metadataTools (symbolHead metaA)))
+                (symbolOrAliasSorts metadataTools (symbolHead metaA)))
         )
     , testCase "getResultSort object"
         (assertEqual ""
             objectS1
             (applicationSortsResult
-                (sortTools metadataTools (symbolHead objectA)))
+                (symbolOrAliasSorts metadataTools (symbolHead objectA)))
         )
     , testCase "getResultSort meta"
         (assertEqual ""
             charListMetaSort
             (applicationSortsResult
-                (sortTools metadataTools (symbolHead metaA)))
+                (symbolOrAliasSorts metadataTools (symbolHead metaA)))
         )
+    , testGroup "subsort" testSubsorts
     ]
   where
     symbolHead symbol = getSentenceSymbolOrAliasHead symbol []
+
+sortA, sortB, sortC, sortD,sortE, sortF, sortG :: Sort Object
+[sortA, sortB, sortC, sortD, sortE, sortF, sortG] =
+    [sortActual [c] [] | c <- "ABCDEFG"]
+
+sortVarR :: Sort Object
+sortVarR = sortVariableSort "R"
+
+testSubsorts :: [TestTree]
+testSubsorts =
+    [test "direct subsort" (isSubsortOf meta sortA sortB)
+    ,test "transitive subsort" (isSubsortOf meta sortA sortC)
+    ,test "not subsort, known sorts" (not (isSubsortOf meta sortD sortE))
+    ,test "not subsort, unknown sorts" (not (isSubsortOf meta sortF sortG))
+    ]
+  where
+    test name cond = testCase name (assertBool "" cond)
+    moduleIndex :: Map.Map ModuleName (KoreIndexedModule ImplicitAttributes)
+    Right moduleIndex = verifyAndIndexDefinition DoNotVerifyAttributes
+        Builtin.koreVerifiers
+        testSubsortDefinition
+    meta :: MetadataTools Object ImplicitAttributes
+    meta = extractMetadataTools $ moduleIndex Map.! testObjectModuleName
+
+
+testSubsortDefinition :: KoreDefinition
+testSubsortDefinition =
+    Definition
+        { definitionAttributes = Attributes []
+        , definitionModules = [ testSubsortModule ]
+        }
+
+testSubsortModule :: KoreModule
+testSubsortModule =
+    Module
+        { moduleName = testObjectModuleName
+        , moduleSentences =
+            [ sortDecl sortA
+            , sortDecl sortB
+            , sortDecl sortC
+            , sortDecl sortD
+            , sortDecl sortE
+            , subsortAxiom sortA sortB
+            , subsortAxiom sortB sortC
+            ]
+        , moduleAttributes = Attributes []
+        }
+  where
+    subsortAxiom :: Sort Object -> Sort Object -> KoreSentence
+    subsortAxiom subSort superSort =
+        constructUnifiedSentence SentenceAxiomSentence
+        SentenceAxiom
```

Comment:
Indentation here does not make sense. SentenceAxiom is probably an argument to constructUnifiedSentence, i.e. it's not at the same level as `constructUnifiedSentence`.

Reply:
I will add the `$` and some parentheses, but `constructUnifiedSentence ConstructorName $ \n...` is the style of almost every other call to `constructUnifiedSentence` in the project, including all several dozen in Test.Kore.Parser.Parser. I'm changing this for consistency with existing code, if we want a bulk style fix to existing code that should be a separate PR.

Reply:
Yes, but there are two issues here:
1. I meant that there should be a way to tell that SentenceAxiom {...} is an argument to constructUnifiedSentence. If it has a `$` in front of it, that may be good enough (though a `$` is not needed here). If it is indented, then it also works. Having both may be the best thing. But here you didn't have any.

2. I think that some of the code using `constructUnifiedSentence` was written before we had any style guide, but I may be wrong.

Reply:
I addressed point 1 as I said I would, making it consistent with other uses by adding a `$` to separate arguments and parenthesizing the `(SentenceAxiom {...})` argument.

For 2. we may very well want to change all that existing code, I just don't want to add another sweeping change to this pull request, and until making that change I would like to be consistent.